pandas/plotting/_matplotlib/core.py

```diff
@@ -115,6 +116,67 @@ def __init__(
         self.sort_columns = sort_columns
 
         self.subplots = subplots
+        if not isinstance(self.subplots, (bool, Iterable)):
+            raise ValueError("subplots should be a bool or an iterable")
+        if isinstance(self.subplots, Iterable):
+
+            supported_kinds = (
+                "line",
+                "bar",
+                "barh",
+                "hist",
+                "kde",
+                "density",
+                "area",
+                "pie",
```

Comment:
I agree it doesn't look good, but it seems to be consistent (i.e. equally bad) as when `subplots=True`. I updated the [notebook](https://gist.github.com/NicolasHug/4d1f60212ba69e6000982951ae47f203) at the end where you can see how it looks. So maybe it can be kept? No strong opinion!

Reply:
i have no strong opinion either, my only concern is for pie chart, if they share the same category, they will have the same color in the pie, but different sizes depending on the corresponding values. 

but no strong opinion